Implementations/CloudFabric.Projections.ElasticSearch/ElasticSearchProjectionRepository.cs

```diff
@@ -299,6 +299,11 @@ public async Task Upsert(Dictionary<string, object?> document, string partitionK
                 }
             );
 
+            if (result.ServerError != null)
+            {
+                throw new Exception($"{result.ServerError.Error}");
```

Comment:
Is there any additional information available about the error? Like index name or something? I would include that into exception to make it easier to troubleshoot.

You can use exception.Data property to store additional information.

Also, there is no need in using string interpolation if the argument is already a string, so $"{...}" is not required.

Reply:
Added full avaliable information into Exception.Data.

About interpolating: ServerError.Error is not a string, but has overrided .ToString() - changed it